### PR TITLE
#58: avoid unnecessary imports

### LINKED ISSUE #58: avoid unnecessary imports
need to go through all files and remove all the unnecessary imports

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #58

### CODE REVIEW

**Comment by cwschilly** [comment]
I only found one unused import: `mpi4py` in `svd_method_of_snapshots.py`. 

While this file uses MPI, it is understood that users would import `mpi4py` whenever they create a `svdMethodOfSnapshots` object (as done in `test_svd_kernel_trick_mpi.py`).

**Comment by fnrizzi** [comment]
why is this PR modifying a lot of stuff and not just the includes?

File: .pylintrc [hunk 1]
```diff
@@ -576,5 +576,5 @@ preferred-modules=
 
 # Exceptions that will emit a warning when being caught. Defaults to
 # "BaseException, Exception".
-overgeneral-exceptions=BaseException,
-                       Exception
+overgeneral-exceptions=bulitins.BaseException,
+                       builtins.Exception
```
Comment (fnrizzi) [inline .pylintrc:580]: this does not belong to this PR

Comment (cwschilly) [reply .pylintrc]: I removed it from the PR

**Review by fnrizzi — APPROVED** [review]
